sstables/sstables.cc

```diff
@@ -975,9 +975,6 @@ future<> sstable::seal_sstable() {
     co_await sstable_write_io_check(rename_file, filename(component_type::TemporaryTOC), filename(component_type::TOC));
     co_await sstable_write_io_check([&] { return dir_f.flush(); });
     co_await sstable_write_io_check([&] { return dir_f.close(); });
-    if (_marked_for_deletion == mark_for_deletion::implicit) {
-        _marked_for_deletion = mark_for_deletion::none;
-    }
```

Comment:
With this change, `sstable::seal_sstable()` should be made private to force all callers to use the api level entry point

Reply:
and maybe even rename the private method to something like `do_seal_sstable`, and change the higher level declaration to `sstable::seal_sstable(bool backup = false)`

Reply:
This PR moves this seal_sstable to filesystem_storage, so only one sstable::seal_sstable() call is left

Reply:
Yeah, I saw that in a later patch...
Still, making this early change will clarify who's calling who, and the move of the low-level seal_sstable (now renamed) to the storage layer will be clearer IMO.